### PR TITLE
Add function to check logs for task complete of fail for each subwallet

### PR DESCRIPTION
**Pull request by b5prolh**
- Actual files 
image

- Users can check their task complete or fail easily by run command `make logs`. **Rewards expect = Task complete * 3 NIM**
image

- The logs will be saved in **my_logs.json** file
image

### CODE REVIEW

File: showlogs.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    # Add each log entry to the table and count statuses
+    for entry in logs:
+        table.add_row([entry['WalletAddr'],entry['CompletedTime'], entry['TrainRuntime'], entry['Status']])
+        if entry['Status'].lower() == "success":
+            success_count += 1
+        elif entry['Status'].lower() == "failed":
+            fail_count += 1
+    nim_expected = success_count * 3
+    # Print the table
+    print(table)
```
Comment (xyq117580) [inline showlogs.py:39]: use print_in_color

File: showlogs.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    print(f"Total tasks: {total_tasks}, Tasks successful: {success_count}, Tasks failed: {fail_count}, Rewards Expected: {nim_expected} $NIM")
```
Comment (xyq117580) [inline showlogs.py:40]: dido

File: execute.py [hunk 3]
```diff
@@ -118,7 +124,7 @@ def register_particle(addr):
     return task['args']
 
 
-def complete_task(wallet_address, max_retries=5, retry_delay=10):
+def complete_task(wallet_address,train_run_time=0, max_retries=5, retry_delay=10):
```
Comment (xyq117580) [inline execute.py:127]: rename train_run_time  to training_duration

File: showlogs.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    """
+    try:
+        # Open the file and load the data
+        with open(file_path, 'r') as file:
+            logs = json.load(file)
+    except FileNotFoundError:
+        print("No log file found. Please make sure the log data has been created.")
+        return
+    except json.JSONDecodeError:
+        print("Error reading the log file. Please check the file format.")
```
Comment (xyq117580) [inline showlogs.py:19]: same here

File: showlogs.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    and prints out the summary statistics of task statuses.
+    
+    :param file_path: Path to the JSON log file.
+    """
+    try:
+        # Open the file and load the data
+        with open(file_path, 'r') as file:
+            logs = json.load(file)
+    except FileNotFoundError:
+        print("No log file found. Please make sure the log data has been created.")
```
Comment (xyq117580) [inline showlogs.py:16]: same here

**Review by xyq117580 — COMMENTED** [review]
Could you share 2 screenshots
1. The result of `make logs`
2. The actual files by `ls -la`

File: showlogs.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    fail_count = 0
+
+    # Add each log entry to the table and count statuses
+    for entry in logs:
+        table.add_row([entry['WalletAddr'],entry['CompletedTime'], entry['TrainRuntime'], entry['Status']])
+        if entry['Status'].lower() == "success":
+            success_count += 1
+        elif entry['Status'].lower() == "failed":
+            fail_count += 1
+    nim_expected = success_count * 3
```
Comment (xyq117580) [inline showlogs.py:37]: nit: 3 is hardcoded. Better put it into param.

**Comment by b5prolh** [comment]
> Could you share 2 screenshots
> 
> 1. The result of `make logs`
> 2. The actual files by `ls -la`

Already update, pls review again. Thanks

File: showlogs.py (hunk 6 above)
Comment (nyeates) [reply showlogs.py:38]: I had thought same thing. Amount might change over time. Or other weird situations

**Comment by nyeates** [comment]
These changes on the var names look good.
The color, meh, whatever. Sometimes the color codes get in the way with raw text format. Its fine.

I wouldnt want to consider upstreaming these until they have been in testing on a few machines for a few days or a week. Once I can figure out how to get the changes over to my modded fork, ill start testing in prod and report back.

**Comment by xyq117580** [comment]
Will take a look at @nyeates 's data points before merging

File: showlogs.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    except FileNotFoundError:
+        print_in_color("No log file found. Please make sure the log data has been created.", "\033[31m")
+        return
+    except json.JSONDecodeError:
+        print_in_color("Error reading the log file. Please check the file format.", "\033[31m")
+        return
+    
+    # Create a PrettyTable object with headers
+    table = PrettyTable()
+    table.field_names = ["WalletAddr","CompletedTime", "TrainRuntime", "Status"]
```
Comment (kezeng2024) [inline showlogs.py:25]: add time units for "TrainRuntime"?

**Comment by nyeates** [comment]
> Will take a look at @nyeates 's data points before merging

Screenshot 2024-05-15 at 1 45 49 AM

One machine of mine has shown it works perfectly over last few days.

This screenshot shows two completely separate methods of logging tasks locally. The top one is my own where I have been semi manually spitting raw output of the miners to files and then using a command to grep them for completions. The bottom one is the table we see here from this code. 

The two are exactly the same +1

**Comment by xyq117580** [comment]
great! Thanks @nyeates for testing! 

I ll merge it tmr after making an announcement just for other people to get prepared about these sudden files. 

I ll tag @b5prolh @nyeates for appreciation. Thanks again!

**Pull request merged**